README
This archive contains the distribution Alien-LibMagic,
version 0.002_01:

  Alien package for the libmagic library

This software is copyright (c) 2014 by Zakariyya Mughal.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


This README file was generated by Dist::Zilla::Plugin::Readme v6.012.


Changes
0.002_01 2020-05-07 16:05:28-0500
  - Switch to Alien::Build. See PR
    <https://github.com/zmughal-p5CPAN/p5-Alien-LibMagic/pull/8>.

  - Use HTTPS to retrieve tarball instead of FTP.
    Thanks to thibaultduponchelle! See
    <https://github.com/zmughal-p5CPAN/p5-Alien-LibMagic/pull/6>.

0.002 2016-04-17 13:09:07-0500
  - Added a minimum version version for Alien::Base `configure_requires`.
    Thanks to Graham Ollis (PLICEASE++)!

0.001 2015-01-24 15:25:24-0600
  - first version


Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v6.012.
use strict;
use warnings;

use 5.006;

use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Alien package for the libmagic library",
  "AUTHOR" => "Zakariyya Mughal <zmughal\@cpan.org>",
  "BUILD_REQUIRES" => {
    "Alien::Build" => "0.32",
    "Alien::Build::MM" => "0.32",
    "ExtUtils::MakeMaker" => "6.52"
  },
  "CONFIGURE_REQUIRES" => {
    "Alien::Build" => "0.32",
    "Alien::Build::MM" => "0.32",
    "ExtUtils::MakeMaker" => "6.52"
  },
  "DISTNAME" => "Alien-LibMagic",
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.006",
  "NAME" => "Alien::LibMagic",
  "PREREQ_PM" => {
    "Alien::Base" => "0.038",
    "Path::Tiny" => 0,
    "parent" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "Module::Load" => 0,
    "Test::More" => 0
  },
  "VERSION" => "0.002_01",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "Alien::Base" => "0.038",
  "Alien::Build" => "0.32",
  "Alien::Build::MM" => "0.32",
  "ExtUtils::MakeMaker" => "6.52",
  "Module::Load" => 0,
  "Path::Tiny" => 0,
  "Test::More" => 0,
  "parent" => 0,
  "strict" => 0,
  "warnings" => 0
);


# BEGIN code inserted by Dist::Zilla::Plugin::AlienBuild
use Alien::Build::MM;
my $abmm = Alien::Build::MM->new;
%WriteMakefileArgs = $abmm->mm_args(%WriteMakefileArgs);
# END code inserted by Dist::Zilla::Plugin::AlienBuild


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);


# BEGIN code inserted by Dist::Zilla::Plugin::AlienBuild
{ package
    MY;
  sub postamble {
    $abmm->mm_postamble(@_);
  }
  sub install {
    $abmm->can('mm_install')
      ? $abmm->mm_install(@_)
      : shift->SUPER::install(@_);
  }
}
# END code inserted by Dist::Zilla::Plugin::AlienBuild



lib/Alien/LibMagic.pm
package Alien::LibMagic;
$Alien::LibMagic::VERSION = '0.002_01'; # TRIAL

$Alien::LibMagic::VERSION = '0.00201';use strict;
use warnings;

use Path::Tiny;
use parent 'Alien::Base';

sub cflags {
	my ($self) = @_;
	my $top_include = File::Spec->catfile( File::Spec->rel2abs($self->dist_dir), qw(include) );
	return "-I$top_include";
}

sub libs {
	my ($self) = @_;
	my $top_lib = File::Spec->catfile( File::Spec->rel2abs($self->dist_dir), qw(lib) );
	my $la_file = path( File::Spec->catfile( $top_lib, 'libmagic.la' ) );
	my ($deps) = $la_file->slurp_utf8 =~ /^dependency_libs=' (.*)'$/m;
	return "-L$top_lib -lmagic $deps";
}

sub Inline {
	my ($self, $lang) = @_;

	if( $lang eq 'C' ) {
		my $params = Alien::Base::Inline(@_);
		$params->{MYEXTLIB} .= ' ' .
			join( " ",
				map { File::Spec->catfile(
					File::Spec->rel2abs($self->dist_dir),
					'lib',  $_ ) }
				qw(libmagic.a)
			);
		# Use static linking instead of dynamic linking on macOS.
		if( $^O eq 'darwin' ) {
			$params->{LIBS} =~ s/-lmagic//g;
		}
		return $params;
	}
}

sub inline_auto_include {
	return  [ 'magic.h' ];
}

1;

=pod

=encoding UTF-8

=head1 NAME

Alien::LibMagic - Alien package for the libmagic library

=head1 VERSION

version 0.002_01

=head1 Inline support

This module supports L<Inline's with functionality|Inline/"Playing 'with' Others">.

=head1 SEE ALSO

L<file(1)>, L<file command|http://darwinsys.com/file/>, L<File::LibMagic>

=head1 AUTHOR

Zakariyya Mughal <zmughal@cpan.org>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2014 by Zakariyya Mughal.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut

__END__
# ABSTRACT: Alien package for the libmagic library



inc/Alien/LibMagic/ModuleBuild.pm
package Alien::LibMagic::ModuleBuild;

use strict;
use warnings;

use parent 'Alien::Base::ModuleBuild';
use Archive::Extract;
use File::Copy;
use File::Spec;
use Module::Load;
use File::chdir;

sub alien_library_destination {
	my ($self) = @_;
	my $dest = $self->SUPER::alien_library_destination;
	if( $^O eq 'MSWin32' ) {
		$dest =~ s,\\,/,g;
	}
	return $dest;
}

sub alien_do_commands {
	my ($self, $phase) = @_;
	my $dir = $self->config_data( 'working_directory' );

	if( $^O eq 'MSWin32' ) {
		load 'Alien::MSYS';
		$self->alien_build_commands([
			map {
				my $c = $_;
				$c =~ s/^%pconfigure/sh %pconfigure/;
				$c
			} @{$self->alien_build_commands}
		]);
		my $extract_to = $phase eq 'build'
			? $dir
			: $self->alien_library_destination;


		my $inc_dir = File::Spec->catfile( $dir, '..', '..', 'inc' );
		my @bins = qw( libgnurx-0.dll libgnurx.dll.a libregex.a );
		if( $phase eq 'build' ) {
			my $archive_file = File::Spec->catfile( $inc_dir, 'mingw-libgnurx-2.5.1-src.tar.gz' );
			my $e = Archive::Extract->new( archive => $archive_file );
			$e->extract( to => $extract_to );
			my $patch_file = File::Spec->catfile( $inc_dir, 'patch\mingw-libgnurx-2.5.1' );
			my $gnurx_dir = File::Spec->catfile( $extract_to, "mingw-libgnurx-2.5.1");
			{
				local $CWD = $gnurx_dir;
				Alien::MSYS::msys( sub  {
					system(qw(patch --binary -i), $patch_file);
					system(qw(sh .\configure));
					system(qw(make));
					move($_, $extract_to ) for ( @bins, 'regex.h' );
					system(qw(make distclean));
				} );
			}

			# make libgnurx available to libfile build
			$ENV{LDFLAGS} //= " "; $ENV{CPPFLAGS} //= " "; # space required on windows
			(my $extract_to_forward = $extract_to) =~ s,\\,/,g;
			#$extract_to_forward =~ s,(\w):/,/$1/,; # drive letter: C:/ -> /C/
			$ENV{LDFLAGS} .= " -L$extract_to_forward";
			$ENV{CPPFLAGS} .= " -I$extract_to_forward";
			$ENV{PATH} .= ";$extract_to_forward";
			print "$ENV{LDFLAGS}\n$ENV{CPPFLAGS}\n$ENV{PATH}\n";
		} elsif( $phase eq 'install' ) {
			my $target = $self->SUPER::alien_library_destination;
			for my $bin (@bins) {
				my $under = ( $bin =~ /\.dll$/ ) ? "bin" : "lib";
				my $target_dir = File::Spec->catfile($target, $under);
				copy(File::Spec->catfile($dir, $bin), $target_dir );
			}
		}
		Alien::MSYS::msys( sub  {
			$self->SUPER::alien_do_commands($phase);
		} );
	} else {
		$self->SUPER::alien_do_commands($phase);
	}
}

1;


lib/Alien/LibMagic/Install/Files.pm
package Alien::LibMagic::Install::Files;
$Alien::LibMagic::Install::Files::VERSION = '0.001';
# allows other packages to use ExtUtils::Depends like so:
#   use ExtUtils::Depends;
#   my $p = new ExtUtils::Depends MyMod, Alien::LibMagic;
# and their code will have all LibMagic available at C level

use strict;
use warnings;

use Alien::LibMagic qw(Inline);
BEGIN { *Inline = *Alien::LibMagic::Inline }
sub deps { () }

__END__

=pod

=encoding UTF-8

=head1 NAME

Alien::LibMagic::Install::Files

=head1 VERSION

version 0.001

=head1 AUTHOR

Zakariyya Mughal <zmughal@cpan.org>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2014 by Zakariyya Mughal.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut


Build.PL

# This file was automatically generated by Dist::Zilla::Plugin::Alien v0.016.
use strict;
use warnings;

use Module::Build 0.28;
use lib qw{inc}; use Alien::Base::ModuleBuild;

my %module_build_args = (
  "alien_build_commands" => [
    "%c --prefix=%s",
    "make"
  ],
  "alien_inline_auto_include" => [],
  "alien_name" => "LibMagic",
  "alien_repository" => {
    "host" => "ftp.astron.com",
    "location" => "/pub/file/",
    "pattern" => "^file-([\\d\\.]+)\\.tar\\.gz\$",
    "protocol" => "ftp"
  },
  "build_requires" => {},
  "configure_requires" => {
    "Alien::Base" => "0.005",
    "File::ShareDir" => "1.03",
    "perl" => "5.010"
  },
  "dist_abstract" => "Alien package for the libmagic library",
  "dist_author" => [
    "Zakariyya Mughal <zmughal\@cpan.org>"
  ],
  "dist_name" => "Alien-LibMagic",
  "dist_version" => "0.001",
  "license" => "perl",
  "module_name" => "Alien::LibMagic",
  "recommends" => {},
  "recursive_test_files" => 1,
  "requires" => {
    "Alien::Base" => "0.005",
    "File::ShareDir" => "1.03",
    "parent" => 0,
    "perl" => "5.006",
    "strict" => 0,
    "warnings" => 0
  },
  "script_files" => [],
  "test_requires" => {
    "Module::Load" => 0,
    "Test::More" => 0,
    "perl" => "5.006"
  }
);


my %fallback_build_requires = (
  "Module::Load" => 0,
  "Test::More" => 0,
  "perl" => "5.006"
);


unless ( eval { Module::Build->VERSION(0.4004) } ) {
  delete $module_build_args{test_requires};
  $module_build_args{build_requires} = \%fallback_build_requires;
}

if ( $^O eq 'MSWin32' ) {
	$module_build_args{requires}{'Alien::MSYS'} = '0';
}

my $build = Alien::Base::ModuleBuild->new(%module_build_args);

$build->create_build_script;
